domain-model.md

```diff
@@ -0,0 +1,40 @@
+## Domain Model
+
+#### Standard Requirements
+```
+The Cohort Manager should be able to support the following interactions
+
+✔ Create a cohort with a cohort name
+✔ Search for a cohort by cohort name
+✔ Add student to a specific cohort
+✔ Remove a cohort by cohort name
+✔ Remove student from a specific cohort
+✔ Throw errors if student or cohort not found
+
+A cohort should have a list of students. Each student should have a studentID, first name, last name, github username, email.
+```
+--- 
+
+#### Extensions
+```
+✔ Search for student by student ID
+✔ Cohorts have fixed capacity at 24 students. Adding students is not possible beyond the 24 limit.
+✔ Cohorts can't have the same name, and can't exist without a name
+✔ The same student can't exist in multiple cohorts.
+- A student can't be removed from a cohort if it wasn't present in the first place.
+- Search for students by name (first and last) and return all matching results
+
+Your program should be composed of at least two classes
```

Comment:
I think your cohort manager class is doing too much lifting and you should make at least a "student" class to make it better manageable. :)